spec/inputs/http_poller_spec.rb

```diff
@@ -23,158 +23,272 @@
-        expect(subject).to receive(:request_async).with(queue, name, url).once
-      }
-
-      subject.send(:run_once, queue) # :run_once is a private method
+    describe "#run" do
+      it "should run at the specified interval" do
+        expect(Stud).to receive(:interval).with(default_interval).once
+        subject.run(double("queue"))
+      end
     end
-  end
 
-  shared_examples("matching metadata") {
-    let(:metadata) { event[metadata_target] }
+    describe "#run_once" do
+      it "should issue an async request for each url" do
+        default_urls.each { |name, url|
+          normalized_url = subject.send(:normalize_request, url)
+          expect(subject).to receive(:request_async).with(queue, name, normalized_url).once
+        }
 
-    it "should have the correct name" do
-      expect(metadata["name"]).to eql(name)
+        subject.send(:run_once, queue) # :run_once is a private method
+      end
     end
 
-    it "should have the correct url" do
-      expect(metadata["url"]).to eql(url)
-    end
+    describe "normalizing a request spec" do
+      shared_examples "a normalized request" do
+        it "should set the method correctly" do
+          expect(normalized.first).to eql(spec_method.to_sym)
+        end
 
-    it "should have the correct code" do
-      expect(metadata["code"]).to eql(code)
-    end
-  }
+        it "should set the options to the URL string" do
+          expect(normalized[1]).to eql(spec_url)
+        end
 
-  shared_examples "unprocessable_requests" do
-    let(:poller) { LogStash::Inputs::HTTP_Poller.new(settings) }
-    subject(:event) {
-      poller.send(:run_once, queue)
-      queue.pop(true)
-    }
+        it "should to set additional options correctly" do
+          opts = normalized.length > 2 ? normalized[2] : nil
+          expect(opts).to eql(spec_opts)
+        end
+      end
 
-    before do
-      allow(poller).to receive(:handle_failure).and_call_original
-      allow(poller).to receive(:handle_success)
-      event # materialize the subject
-    end
+      let(:normalized) { subject.send(:normalize_request, url) }
 
-    it "should enqueue a message" do
-      expect(event).to be_a(LogStash::Event)
-    end
+      describe "a string URL" do
+        let(:url) { "http://localhost:3000" }
+        let(:spec_url) { url }
+        let(:spec_method) { :get }
+        let(:spec_opts) { nil }
 
-    it "should enqueue a message with '_http_request_failure' set" do
-      expect(event["_http_request_failure"]).to be_a(Hash)
-    end
+        include_examples("a normalized request")
+      end
 
-    it "should tag the event with '_http_request_failure'" do
-      expect(event["tags"]).to include('_http_request_failure')
+      describe "URL specs" do
+        context "with basic opts" do
+          let(:spec_url) { "http://localhost:3000" }
+          let(:spec_method) { "post" }
+          let(:spec_opts) { {:"X-Bender" => "Je Suis Napoleon!"} }
+
+          let(:url) {
```

Comment:
ehhhh. EHHHHH.

I'm on the fence, but historically I've always had multiline blocks be `do .. end` even if they intend to return (even implicitly) a value; a similar example outside rspec:

```ruby
["google.com", "twitter.com"].collect do |host|
   addresses = Resolv.new.getaddresses(host)
   addresses.shuffle.first
done
```

(I know the above could be oneline, but humor me.

In rspec, I'd write thsi similarly:
```ruby
let(:hosts) { [ "google.com", "twitter.com" ] }
let (:random_address) do
   addresses = Resolv.new.getaddresses(hosts.shuffle.first)
   addresses.shuffle.first
end
```